load.c

```diff
@@ -1208,7 +1225,7 @@ require_internal(rb_execution_context_t *ec, VALUE fname, int exception, bool wa
                 result = TAG_RETURN;
             }
             else if (RTEST(rb_hash_aref(realpaths,
-                                        realpath = rb_realpath_internal(Qnil, path, 1)))) {
+                                        realpath = rb_realpath_internal_cached(realpath_map, path)))) {
```

Comment:
If `realpath_map` is updated here, I think the `rb_hash_aset` call that updates it later on in this function (around line 1271) should be removed. The `rb_hash_aset` to update `realpaths` will still be needed.

Reply:
`realpath_map` is indeed updated in `realpath_internal_cached`, removed the redundant update around line 1271.